core/src/main/java/org/infinispan/notifications/cachelistener/cluster/RemoteClusterListener.java

```diff
@@ -35,7 +35,7 @@
  * @since 7.0
  */
 @ThreadSafe
-@Listener(primaryOnly = true)
+@Listener(primaryOnly = true, conditionType = Listener.ConditionType.ONLY_POST)
```

Comment:
I think we also need to have a sanity check in CacheNotifier.addListener to prevent registering clustered listener that are not ConditionType.ONLY_POST